### PR TITLE
Jdrempel/test loader menu

### LINKED ISSUE #9: Create a TestLoaderMenu
There needs to be a menu that prompts the user to select a test from a list.

Basic functionality:

1. User enters TestMenu
2. User selects `1. Load Test`
3. TestLoaderMenu displays all the tests in `TestLoader.library` (i.e. `DynamicImporter.class_list`).
3.a) If more tests exist than can be displayed reasonably at one time, implement pagination
4. User selects a test (implementation could be involve typing a number index or the name of the test)
5. `TestMenu` loads the appropriate test
6. User is returned to TestMenu from TestLoaderMenu

### PR DESCRIPTION
**Pull request by jdrempel**
**Summary**
- Adds a new TestLoaderMenu class for selecting a test to run
- Removes old test loading code
- Allows Prompts to handle "any" type of input, i.e. it can accept either a `str` or a `int` if the correct parameter is given
- Disables the serial connection in `Test.__init__()` for now

closes #9 

**Other Notes**
- It would be good to have the serial connection in `Test.__init__()` be enabled (obviously) - perhaps the way to do this is with a try..except that catches exceptions related to the serial line not existing.

### CODE REVIEW

File: src/menus/test_loader_menu.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        test_path = path.join(path.dirname(path.abspath(__file__)), "..", "testdata")
+        files = listdir(test_path)
+
+
+        for f in files:
+            if f in ["test_module.py", "test_registry.py", "connections.py"]:  # blacklist of py files
+                continue
+            name = "".join(f.split(".")[:-1])
+            if name not in ["", "\n", "\r", "\r\n"] and f.endswith(".py"):
+                self.prompt_options[len(self.prompt_options.keys())+1] = f"{name.title()}Test"
```
Comment (Lars-Kish) [inline src/menus/test_loader_menu.py]: Is this just appending to a list? Could it be done in a cleaner way?

File: src/testdata/test_module.py [hunk 2]
```diff
@@ -24,8 +24,9 @@ class Test(ABC):
     """
 
     def __init__(self):
-        self.connection = SerialLine(f"Serial-{self.__class__.__name__}", "/dev/ttyUSB0", 9600)
-        self.connection.open()  # TODO: Lock the line for at least a second before allowing any tx
+        pass
```
Comment (Lars-Kish) [inline src/testdata/test_module.py:27]: I think a try-catch would be good here

File: src/menus/test_loader_menu.py (hunk 1 above)
Comment (jdrempel) [reply src/menus/test_loader_menu.py]: Yep, definitely.

File: src/testdata/test_module.py (hunk 2 above)
Comment (jdrempel) [reply src/testdata/test_module.py:27]: Agreed. Will take a look at this.

**Review by jdrempel — COMMENTED** [review]
Updated the dict modification to use `dict.update()`.

**Review by Lars-Kish — APPROVED** [review]

**Pull request merged**